File: Cribbage/AIs/Pegger/RandomPeggerAI.cpp
#include "RandomPeggerAI.h"

#include "PeggerAI.h"
#include "../../Utils/Random.h"
#include "../../Utils/Card.h"
#include "../../Utils/Hand.h"

RandomPeggerAI::~RandomPeggerAI() {}

Card RandomPeggerAI::select(const std::vector<Card>& played, const std::vector<Card>& avail)
{
    std::vector<Card> validCards = PeggerAI::valid(played, avail);
    if (validCards.size() == 0) { return Card(); }

    int i = Random::Instance()->randInt() % validCards.size();
    return validCards[i];
}

File: CribbageFlutter/cribbage/lib/models/ai/pegger/random_pegger_ai.dart
import 'dart:math';
import './pegger_ai.dart';
import '../../deck/card.dart';

class RandomPeggerAI implements PeggerAI {
  final Random _rand = Random();

  Card? select(List<Card> played, List<Card> avail) {
    List<Card> validCards = PeggerAI.valid(played, avail);
    if (validCards.length == 0) {
      return null;
    }

    int i = _rand.nextInt(validCards.length);
    return validCards[i];
  }
}
